src/ethereum_test_specs/eof.py

```diff
@@ -265,6 +299,67 @@ def verify_result(self, result: CompletedProcess, expected_result: Result, code:
                     got=f"{actual_exception} ({actual_message})",
                 )
 
+    def generate_eof_contract_create_transaction(self) -> Transaction:
+        """Generate a transaction that creates a contract."""
+        assert self.sender is not None, "sender must be set to generate a StateTest."
+        assert self.post is not None, "post must be set to generate a StateTest."
+
+        initcode: Container
+        deployed_container: Container | None
+        if self.container_kind == ContainerKind.INITCODE:
+            if "deployed_container" in self.model_fields_set:
+                # In the case of an initcontainer where we know the deployed container,
+                # we can use the initcontainer as-is.
+                initcode = self.container
+                deployed_container = self.deployed_container
+            else:
+                # The contract will be deployed as a EOF factory contract.
+                deployed_container = Container(
+                    sections=[
+                        Section.Code(Op.EOFCREATE[0](0, 0, 0, 0) + Op.STOP),
+                        Section.Container(self.container),
+                    ]
+                )
+                initcode = Container(
+                    sections=[
+                        Section.Code(Op.RETURNCONTRACT[0](0, 0)),
+                        Section.Container(deployed_container),
+                    ]
+                )
+        else:
+            initcode = Container(
+                sections=[
+                    Section.Code(Op.RETURNCONTRACT[0](0, 0)),
+                    Section.Container(self.container),
+                ]
+            )
+            deployed_container = self.container
+
+        tx = Transaction(
+            sender=self.sender,
+            to=None,
+            gas_limit=10_000_000,
+            data=initcode,
+        )
+
+        if self.expect_exception is not None or deployed_container is None:
```

Comment:
the condition `deployed_container is None`: do I understand correctly, this will only trigger when `deployed_container=None` is set explicitly on the `eof_test`, and it is in `INITCODE` mode _and_ there was no `expect_exception` set, that is to say, deployment fails, but for reasons other than eof validation?

If so, maybe this could use some documentation, at least expand the one in the PR description

Reply:
I added docstrings to explain the behavior of each field:
https://github.com/ethereum/execution-spec-tests/blob/fc6e7016f245b24bf17c749b2c7dcf77690c36c3/src/ethereum_test_specs/eof.py#L145-L231

These can be seen in the documentation: https://ethereum.github.io/execution-spec-tests/main/library/ethereum_test_specs/#ethereum_test_specs.EOFTest (Will get updated once this PR is merged)

The documentation itself could be improved to list `EOFTest` as a spec type that can be used with some examples, but we can leave that for a follow up PR.